src/modules/avatar/controller/inputs/keyboardInput.ts

```diff
@@ -127,31 +129,39 @@ export class KeyboardInput implements IInputHandler {
         }
     }
 
-    private _onKeyDown(evt: ActionEvent):void {
-        if (!evt.sourceEvent?.code) {
+    private _onKeyDown(event: ActionEvent): void {
+        const sourceEvent = event.sourceEvent as Event;
+        if (!sourceEvent || !(sourceEvent instanceof KeyboardEvent) || !sourceEvent.code) {
             return;
         }
 
-        this._inputMap[evt.sourceEvent.code] = evt.sourceEvent.type === "keydown";
-        this._shiftKey = evt.sourceEvent.shiftKey === true;
+        this._inputMap[sourceEvent.code] = sourceEvent.type === "keydown";
+        this._shiftKey = sourceEvent.shiftKey === true;
```

Comment:
`=== true` here and elsewhere looks suspicious: if the variable on the left is a `boolean` then don't need to test it against `true`.